bench/Autofac.Benchmarks/LambdaResolveBenchmark.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Autofac.Core;
+
+namespace Autofac.Benchmarks;
+
+public class LambdaResolveBenchmark
+{
+    private IContainer _container;
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        var builder = new ContainerBuilder();
+
+        builder.RegisterType<MyDependency1>();
+
+        // The original componentcontext way.
+        builder.Register((ctxt) => new MyComponent(ctxt.Resolve<MyDependency1>(), "arg"))
+               .As<IServiceExistingMethodCapturedArg>();
+
+        // Capture the argument directly.
+        builder.Register((MyDependency1 dep1) => new MyComponent(dep1, "arg"))
+               .As<IServiceNewMethodCapturedArg>();
+
+        // New parameter-supporting method, but the argument comes from a resolve parameter.
+        builder.Register((MyDependency1 dep1, string arg1) => new MyComponent(dep1, arg1))
+               .As<IServiceNewMethodOpenArg>();
+
+        // Check against the reflection way as well.
+        builder
+           .RegisterType<MyComponent>()
+           .UsingConstructor(typeof(MyDependency1), typeof(string))
+           .WithParameter(TypedParameter.From("arg"))
+           .As<IServiceFromReflection>();
+
+        _container = builder.Build();
+    }
+
+    [Benchmark]
+    public object ResolveManuallyWithComponentContext()
+    {
+        return _container.Resolve<IServiceExistingMethodCapturedArg>();
```

Comment:
I kinda like this over the `GC.KeepAlive` we have in other tests. I think we had `GC.KeepAlive` from before when BenchmarkDotNet supported returning objects.